cuda_core/cuda/core/experimental/_memory.pyx

```diff
@@ -906,3 +911,496 @@ class _SynchronousMemoryResource(MemoryResource):
+        all steps the function took are rolled back so a new allocation is not created.
+
+        Parameters
+        ----------
+        buf : Buffer
+            The existing buffer to grow
+        new_size : int
+            The new total size for the allocation
+        config : VirtualMemoryResourceOptions, optional
+            Configuration for the new physical memory chunks. If None, uses current config.
+
+        Returns
+        -------
+        Buffer
+            The same buffer with updated size and properties, preserving the original pointer
+        """
+        if config is not None:
+            self.config = config
+
+        # Build allocation properties for new chunks
+        prop = driver.CUmemAllocationProp()
+        prop.type = VirtualMemoryResourceOptions._allocation_type_to_driver(self.config.allocation_type)
+        prop.location.type = VirtualMemoryResourceOptions._location_type_to_driver(self.config.location_type)
+        prop.location.id = self.device.device_id
+        prop.allocFlags.gpuDirectRDMACapable = 1 if self.config.gpu_direct_rdma else 0
+        prop.requestedHandleTypes = VirtualMemoryResourceOptions._handle_type_to_driver(self.config.handle_type)
+
+        # Query granularity
+        gran_flag = VirtualMemoryResourceOptions._granularity_to_driver(self.config.granularity)
+        res, gran = driver.cuMemGetAllocationGranularity(prop, gran_flag)
+        raise_if_driver_error(res)
+
+        # Calculate sizes
+        additional_size = new_size - buf.size
+        if additional_size <= 0:
+            # Same size: only update access policy if needed; avoid zero-sized driver calls
+            descs = self._build_access_descriptors(prop)
+            if descs:
+                res, = driver.cuMemSetAccess(int(buf.handle), buf.size, descs, len(descs))
+                raise_if_driver_error(res)
+            return buf
+
+        aligned_additional_size = VirtualMemoryResource._align_up(additional_size, gran)
+        total_aligned_size = VirtualMemoryResource._align_up(new_size, gran)
+        aligned_prev_size = total_aligned_size - aligned_additional_size
+        addr_align = self.config.addr_align or gran
+
+        # Try to extend the existing VA range first
+        res, new_ptr = driver.cuMemAddressReserve(
+            aligned_additional_size,
+            addr_align,
+            int(buf.handle) + aligned_prev_size,  # fixedAddr hint - aligned end of current range
+            0
+        )
+
+        if res != driver.CUresult.CUDA_SUCCESS or new_ptr != (int(buf.handle) + aligned_prev_size):
+            # Check for specific errors that are not recoverable with the slow path
+            if res in (driver.CUresult.CUDA_ERROR_INVALID_VALUE, driver.CUresult.CUDA_ERROR_NOT_PERMITTED, driver.CUresult.CUDA_ERROR_NOT_INITIALIZED, driver.CUresult.CUDA_ERROR_NOT_SUPPORTED):
+                raise RuntimeError(f"Failed to extend VA range: {res}")
```

Comment:
Maybe use `raise_if_driver_error(res)` here as well? Would be good to consistently raise the driver error.